### PR TITLE
Switch from mailgun to ses

### PR DESCRIPTION
**Pull request by phildarnowsky**
#### Summary of changes
**Asana Ticket:** [Ops | Mailgun vs Amazon SES to send emails from dotcom](https://app.asana.com/0/555089885850811/1160349191370777)

Here we implement the switch from Mailgun to SES so we don't get charged for Mailgun.

### CODE REVIEW

**Comment by digitalcora** [comment]
About the testing approach: rather than passing in the "perform" function, could we configure it in the application environment? It seems like we'd always want the mailer to be the "mock" version in the test env, and never in others. This would avoid the need to store the function in the `conn` and pass it down through several layers of calls.

File: mix.exs [hunk 1]
```diff
@@ -13,7 +13,7 @@ defmodule DotCom.Mixfile do
       ],
       test_coverage: [tool: ExCoveralls],
       dialyzer: [
-        plt_add_apps: [:mix, :phoenix_live_reload, :laboratory],
+        plt_add_apps: [:mix, :phoenix_live_reload, :laboratory, :ex_aws, :ex_aws_ses],
```
Comment (digitalcora) [inline mix.exs]: I know you were having issues with Dialyzer at first — is this config change still required? If ExAWS is specified as a dependency in the usual way, it _should_ get picked up automatically.

Comment (phildarnowsky) [reply mix.exs]: That's what I thought, but this change, though I was cargo culting a bit, is what got Dialyzer to pass. Still, I will double check without it.

Comment (phildarnowsky) [reply mix.exs]: @digitalcora yup, took them out and dialyzer fails again.

Comment (digitalcora) [reply mix.exs]: I don't suppose I'll ever understand all the mysteries of Dialyzer ¯\\\_(ツ)_/¯

File: apps/feedback/test/mailer_test.exs [hunk 2]
```diff
@@ -54,7 +54,10 @@ defmodule Feedback.MailerTest do
     end
 
     test "uses the comments of the message for the description" do
-      Mailer.send_heat_ticket(%{@base_message | comments: "major issue to report"}, nil)
+      Mailer.send_heat_ticket(
+        %{@base_message | comments: "major issue to report"},
+        nil
+      )
```
Comment (digitalcora) [inline apps/feedback/test/mailer_test.exs]: Looks like this change is now formatting-only? Could revert it to minimize diff noise

File: apps/feedback/test/mailer_test.exs [hunk 3]
```diff
@@ -81,7 +84,11 @@ defmodule Feedback.MailerTest do
     end
 
     test "the email does not have leading or trailing spaces" do
-      Mailer.send_heat_ticket(%{@base_message | email: "   fake_email@gmail.com  "}, nil)
+      Mailer.send_heat_ticket(
+        %{@base_message | email: "   fake_email@gmail.com  "},
+        nil
+      )
```
Comment (digitalcora) [inline apps/feedback/test/mailer_test.exs]: As above

File: mix.exs (hunk 1 above)
Comment (phildarnowsky) [reply mix.exs]: I don't think one _can_ ever understand all the mysteries of Dialyzer. Maybe Joe Armstrong did.

File: apps/feedback/mix.exs [hunk 4]
```diff
@@ -21,7 +21,7 @@ defmodule Feedback.Mixfile do
   #
   # Type "mix help compile.app" for more information
   def application do
-    [applications: [:logger, :mailgun, :timex, :briefly]]
+    [applications: [:logger, :mailgun, :timex, :briefly, :ex_aws, :ex_aws_ses, :mail]]
```
Comment (digitalcora) [inline apps/feedback/mix.exs]: It seems odd that we're manually specifying an `applications` key — normally this list is automatically inferred from the dependencies, and if you need a built-in Elixir application that's not started by default (like `logger`), you put it in `extra_applications`.

Comment (phildarnowsky) [reply apps/feedback/mix.exs]: perhaps, but given the hard deadline let's look into this later, once we've confirmed successful transition to SES.

**Pull request merged**